src/sessions.js

```diff
@@ -0,0 +1,137 @@
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+import { listen } from '@nextcloud/notify_push'
+import { sessionApi } from './services/SessionApi.js'
+import store from './store/main.js'
+import axios from '@nextcloud/axios'
+
+const SESSION_INTERVAL = 90 // in seconds
+
+let hasPush = false
+
+/**
+ * used to verify, whether an event is originated by ourselves
+ *
+ * @param token
+ */
+function isOurSessionToken(token) {
+	if (axios.defaults.headers['x-nc-deck-session']
+		&& axios.defaults.headers['x-nc-deck-session'].startsWith(token)) {
+		return true
+	} else {
+		return false
+	}
+}
+
+hasPush = listen('deck_board_update', (name, body) => {
+	// ignore update events which we have triggered ourselves
+	if (isOurSessionToken(body._causingSessionToken)) return
+
+	// only handle update events for the currently open board
+	const currentBoardId = store.state.currentBoard?.id
+	if (body.id !== currentBoardId) return
+
+	store.dispatch('refreshBoard', currentBoardId)
+})
+
+/**
+ * is the notify_push app active and can
+ * provide us with real time updates?
+ */
+export function isNotifyPushEnabled() {
+	return hasPush
+}
+
+/**
+ *
+ * @param boardId
+ */
+export function createSession(boardId) {
+
+	if (!isNotifyPushEnabled()) {
+		// return a dummy object
+		return {
+			async close() {},
+		}
+	}
+
+	// let's try to make createSession() synchronous, so that
+	// the component doesn't need to bother about the asynchronousness
+	let tokenPromise
+	let token
+	const create = () => {
+		tokenPromise = sessionApi.createSession(boardId).then(res => res.token)
+		tokenPromise.then((t) => {
+			token = t
+			axios.defaults.headers['x-nc-deck-session'] = t
+		})
+	}
+	create()
+
+	const ensureSession = async () => {
+		if (!tokenPromise) {
+			create()
+			return
+		}
+		try {
+			await sessionApi.syncSession(boardId, await tokenPromise)
+		} catch (err) {
+			// session probably expired, let's
+			// create a fresh session
+			create()
+		}
+	}
+
+	// periodically notify the server that we are still here
+	let interval = setInterval(ensureSession, SESSION_INTERVAL * 1000)
+
+	// close session when tab gets hidden/inactive
+	const visibilitychangeListener = () => {
+		if (document.visibilityState === 'hidden') {
+			sessionApi.closeSessionViaBeacon(boardId, token)
+			tokenPromise = null
+			token = null
+			delete axios.defaults.headers['x-nc-deck-session']
+
+			// stop session refresh interval
+			clearInterval(interval)
+		} else {
+			// tab is back in focus or was restored from the bfcache
+			ensureSession()
+
+			// we must assume that the websocket connection was
+			// paused and we have missed updates in the meantime.
+			store.dispatch('refreshBoard', store.state.currentBoard?.id)
+
+			// restart session refresh interval
+			interval = setInterval(ensureSession, SESSION_INTERVAL * 1000)
+		}
+	}
+	document.addEventListener('visibilitychange', visibilitychangeListener)
```

Comment:
closing the session when the tab is not active might be a bit confusing, even if it's technically a cool feature. Google Docs also doesn't do this afaik. @juliushaertl / @jancborchardt Do we want this?